__tests__/app.test.js

```diff
@@ -12,6 +14,82 @@ afterAll(() => {
 	return db.end();
 });
 
+/**
+ * Register Custom Matchers
+ */
+expect.extend({
+    toBeValidAPIEndpoint,
+    toBeValidRequestMethod
+});
+
+describe('API', () => {
+
+    test('GET /api should respond with a 200 status code & a body containing an api property', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            expect(response.body).toHaveProperty('api');
+        });
+    });
+
+    test('GET /api response should equal the object stored within our projects endpoints.json file', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            expect(response.body.api).toEqual(apiDesc);
+        });
+    });
+
+    /**
+     * Although these tests are not required, I felt as though there should be a standard that endpoints.js
+     * follows; and at minimum it should include a valid method, a valid endpoint and a description.
+     * These tests would ensure that any typos that invalidate that standard are caught before hitting prod.
+     */
+    describe('Syntactically validate endpoints.json by virtue of analysing the response from /api', () => {
+
+        /**
+         * Uses custom matchers, see ./helpers/custom.js
+         */
+        test('GET /api should respond with a body containing appropriate api endpoints', () => {
+            return request(app)
+            .get('/api')
+            .expect(200)
+            .then((response) => {
+                const endpoints = response.body.api;
+                expect(Object.entries(endpoints).length).toBeGreaterThan(0);
+                // endpointDesc example: "GET /api/example"
+                for (const endpointDesc in endpoints) {
+                    const [ requestType, endpointPath ] = endpointDesc.split(" ");
+                    expect(requestType).toBeValidRequestMethod();
+                    expect(endpointPath).toBeValidAPIEndpoint();
+                }
+            });
+        });
+
+        /**
+         * J.D: This is a bit wordy & possibly hard to follow, TODO: consider potential refactors.
+         */
+        test('GET /api should respond with api endpoints that contain, at minimum, a description', () => {
+            return request(app)
+            .get('/api')
+            .expect(200)
+            .then((response) => {
+                const iterableEndpoints = Object.entries(response.body.api);
+                expect(iterableEndpoints.length).toBeGreaterThan(0);
+                iterableEndpoints.forEach(([endpointKey, endpointValue]) => {
+                    expect(endpointValue).toMatchObject({
+                        description: expect.any(String)
+                    });
+                }); // foreach
+            }); // response
+        }); // test
```

Comment:
Try to remove any unneeded comments from pushed code for presentation (you can keep in those you need to document your code)